src/main/java/org/jabref/model/database/BibDatabases.java

```diff
@@ -13,19 +14,19 @@ private BibDatabases() {
     }
 
     /**
-     * Gets a collection of bibentries and sets an ID for every entry. After that
+     * Gets a collection of bibEntries and sets an ID for every entry. After that
      * all entries will be inserted into a new BibDatabase.
      *
-     * @param bibentries a collection that contains {@link BibEntry}
+     * @param bibEntries a collection that contains {@link BibEntry}
      * @return BibDatabase that contains the entries
      */
-    public static BibDatabase createDatabase(Collection<BibEntry> bibentries) {
+    public static BibDatabase createDatabase(Collection<BibEntry> bibEntries) {
         BibDatabase database = new BibDatabase();
 
-        for (BibEntry entry : bibentries) {
+        for (BibEntry entry : bibEntries) {
```

Comment:
I might be mistaken but I think `insertEntries` already takes care of the id generation.

Reply:
`insertEntries` add existing IDs to the database, but does not set them.

Reply:
However, I just noticed that `BibDatabases.createDatabase` is only called outside of test in `ParserResult`. Also, all `BibEntry` constructors assign an ID to each `BibEntry`. Is it possible that the IDs are being changed for something that relates to parsing?

Reply:
No, IDs should be fixed upon creation and are never changed. Moreover, they are only used in the connection with shared databases to identify remote entries with local ones. Thus, I guess this method can be removed.

Reply:
It looks like this method is only used in `ParserResult` (and test). Perhaps we can remove the method entirely. In addition, based on what you said about IDs not changing, it might be worth it to remove the `setId` method from `BibEntry`, which is only called in `BibDatabases.createDatabase` and test. Maybe that could be in another PR. Thoughts?

Reply:
Sounds good! Do you want to work on this, too? Then I let you decide if you want to implement it as part of this PR or a new one.

Reply:
Let's do it as a new PR once I've finished this one and the DBMS synchronization.